assembly/fabric8-ide-gwt-app/src/main/resources/org/eclipse/che/public/keycloak/RhCheKeycloak.js

```diff
@@ -406,103 +456,134 @@ function initAnalytics(writeKey){
-                if (lastProvisioningDate) {
-                    var w = window.open('', 'osio_provisioning');
-                    w && w.close();
+            const segmentWriteKeyUrl = '/api/fabric8-che-analytics/segment-write-key';
+            get(segmentWriteKeyUrl)
+            .then(function (request) {
+                var segmentKey = request.responseText;
+                if (segmentKey!= '') {
+                    initAnalytics(segmentKey);
                 }
-                identifyUser(keycloak)
-                .then(function() {
-                    if (window.analytics && lastProvisioningDate) {
-                        analytics.track(telemetry_event_provision_user_for_che);
+                
+                var promise = originalInit(initOptions);
+                promise.success(function(arg) {
+                    var keycloak = kc;
+                    identifyUserWithId(keycloak);
+                    var lastProvisioningDate = sessionStorage.getItem('osio-provisioning');
+                    var lastProvisioningTimeoutFailure = sessionStorage.getItem('osio-provisioning-timeout-failure');
+                    sessionStorage.removeItem('osio-provisioning');
+                    sessionStorage.removeItem('osio-provisioning-timeout-failure');
+                    if (lastProvisioningDate) {
+                        var w = window.open('', 'osio_provisioning');
+                        w && w.close();
                     }
-                    return performAccounkLinking(keycloak);
-                })
-                .then(()=>{
-                    return setUpNamespaces(keycloak);
-                })
-                .then(() => {
-                    if (window.analytics && isInCheDashboard) {
-                        analytics.track(telemetry_event_enter_che_dashboard);
+                    if (lastProvisioningDate || lastProvisioningTimeoutFailure) {
+                        track(telemetry_event_provision_user_for_che);
+                    }
+                    performAccounkLinking(keycloak)
+                    .then(()=>{
+                        return setUpNamespaces(keycloak);
+                    })
+                    .then(() => {
+                        identifyUser(keycloak)
+                        .then(function() {
+                            if (isInCheDashboard) {
+                                track(telemetry_event_enter_che_dashboard);
+                            }
+                            setStatusMessage(osio_msg_started);
+                            finalPromise.setSuccess(arg);
+                        })
+                    })
+                    .catch((errorMessage) => {
+                        post('/api/fabric8-che-analytics/error', errorMessage);
+                        setStatusMessage(osio_msg_error_no_resources);
+                        finalPromise.setError({ error: 'invalid_request', error_description: errorMessage });
+                    });
```

Comment:
are you able to use await/async ?
it would make code more compact/maintainable instead of chaining callbacls

Reply:
fixed